src/utils/packagePath.ts

```diff
@@ -0,0 +1,55 @@
+import * as path from 'path';
+import { RegistryPackageType } from '../targets/registry';
+import { parseCanonical } from './canonical';
```

Comment:
Let's merge this file with `canonical` module for simplicity (unless there's a reason to keep them separate).